docs/rfc/rfc-3.adoc

```diff
@@ -0,0 +1,119 @@
+:toc: macro
+
+= RFC 3: Rewards
+
+:icons: font
+:numbered:
+toc::[]
+
+== Overview
+
+When various actions are taken in the tBTC system,
+some TBTC tokens are collected as fees.
+These tokens may be sold for T tokens
+which are then distributed as rewards to various participants,
+or used as rewards directly.
+
+Operators are rewarded for being present and available in the sortition pool,
+and for performing various required transactions on the Ethereum blockchain
+as a member of an active wallet.
+In addition, any Ethereum account may be rewarded
+for performing various necessary maintenance tasks.
+
+=== Sortition pool rewards
+
+All operators present in the sortition pool
+accumulate rewards while they remain in the pool.
+For a simple solution,
+any rewards paid to the sortition pool
+are immediately divided among the currently present operators
+in proportion to their current weight in the pool.
+This requires that joining and leaving the sortition pool quickly is prevented
+so that operators can't pop in for rewards and leave afterwards,
+but requires very little bookkeeping or expensive calculations.
+
+The pool maintains a single global variable for rewards:
+`accumulatedWeightReward`.
+In addition, the total weight of the sortition pool `poolWeight` can be queried.
+Whenever `r` tokens are paid to the pool,
+the accumulator is increased: `accumulatedRewardPerWeight += r / poolWeight`.
+
+Each operator maintains 2 variables for their rewards:
+`weightRewardCollected` and `rewardAvailable`.
+In addition, each operator's `weight` can be queried.
+Whenever an operator's weight changes from `oldWeight` to `newWeight`,
+the rewards earned with the old weight are collected:
+`rewardAvailable += (accumulatedWeightReward - weightRewardCollected) * oldWeight`,
+`weightRewardCollected = accumulatedWeightReward`
+and `weight = newWeight`.
+
+At any time, an operator can withdraw their collected rewards,
+transferring `rewardAvailable` T tokens to them,
+and setting their `rewardAvailable = 0`.
+
+=== Wallet action rewards
+
+Whenever a wallet is required to perform some transaction on-chain,
+a _transaction reward_ is reserved for the wallet.
+When the transaction is correctly submitted,
+the transaction reward is divided
+between the submitter and the rest of the wallet members.
```

Comment:
How do we divide the reward between wallet members? It sounds similar to per-group beacon rewards that we know turned to be problematic because of withdrawal costs. Can we just reward the submitter and if there is anything left (because of a delay penalty, for example), return it to pool-wide rewards pool? 

(similar approach was proposed in https://github.com/keep-network/keep-core/pull/2585)